### PR TITLE
bunch of updates :)

### PR DESCRIPTION
**Pull request by RahavLussato**
1. change componentDidUpdate to componentWillReceiveProps : detection of changes need to happen before the chart will update to avoid multiple updates because the same changes. its more correct way by the best practices of react.

2. add new prop `chartVersion`: in my case i have situations that i making math calc on all the data and change all the data, in this cases i don't want to waste time to go all over the data, i want to clear all the and make a new chart (its much faster).

3. add new prop `activeSkip`: in most of the cases when chart is get new data provider that bigger then the old one, the old data is not changed and i want to handle only the new indexes (again performance issue) so i added activeSkip prop to enable this, and new part that check if the new `dataProvider` is bigger than the old one and activeSkip is turned on it will handle only the part of the data that exists in the new `dataProvider`

4. change ` this.state.chart.validateNow();` to `this.state.chart.validateNow(true, false);`
it was doing errors inside amcharts.

### CODE REVIEW

**Comment by iddan** [comment]
Why isn't it merged? Is it tested?

**Comment by RahavLussato** [comment]
the comment is not valid anymore, i changed more stuff from the time i wrote that.

**Comment by RahavLussato** [comment]
its hard to contribute when its take long time to merge PR

**Comment by Pauan** [comment]
@RahavLussato I'm very sorry for the delay on this. I'll review this pull request now.

File: amcharts3-react.js [hunk 1]
```diff
@@ -173,7 +199,7 @@
           break;
 
         default:
-          if (x !== y) {
+          if (!isEqual(x,y)) {
```
Comment (Pauan) [inline amcharts3-react.js]: The type checks above means that the `default` case will only handle simple types like strings.

Therefore, using `isEqual` is unnecessary and inefficient.

File: amcharts3-react.js [hunk 2]
```diff
@@ -227,15 +262,18 @@
...
+                didUpdateData = true;
+            } else {
+                didUpdateClass = true;
+            }
         }
       }
     }
 
-    return didUpdate;
+    return {didUpdateData:didUpdateData, didUpdateClass:didUpdateClass};
```
Comment (Pauan) [inline amcharts3-react.js]: See the comment below for `validateData`

File: package.json [hunk 3]
```diff
@@ -1,10 +1,13 @@
...
-  "version": "1.1.8",
+  "version": "1.1.9",
   "license": "SEE LICENSE IN LICENSE",
   "main": "index.js",
   "repository": {
     "type": "git",
     "url": "https://github.com/amcharts/amcharts3-react.git"
+  },
+  "dependencies": {
+    "lodash": "^4.17.4"
```
Comment (Pauan) [inline package.json]: I'd rather not have a dependency on `lodash`, especially because it's not needed (see comments above).

File: package.json [hunk 4]
```diff
@@ -1,10 +1,13 @@
 {
   "name": "amcharts3-react",
-  "version": "1.1.8",
+  "version": "1.1.9",
```
Comment (Pauan) [inline package.json]: I prefer to put multiple changes into a single update, so I prefer to bump the version number when releasing, not when making the changes.

File: amcharts3-react.js [hunk 5]
```diff
@@ -1,3 +1,5 @@
+const isEqual = require('lodash/isEqual');
```
Comment (Pauan) [inline amcharts3-react.js]: This is not needed, see comments below.

**Review by Pauan — CHANGES_REQUESTED** [review]
Thank you for your work, but here are some things we need to address before I will merge this.

Also, please do not commit the `node_modules` folder into git.

**Pull request closed without merging**